gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/discord/Listener.java

```diff
@@ -57,9 +49,18 @@ public void onMessageReceived(@NotNull MessageReceivedEvent event) {
         // Check if the category is not null and then get its name
         String categoryName = (category != null) ? category.getName() : "No Category"; // You can change "No Category" to a default value if needed
 
+        // check if author is gendox user and if not, create new user
+        if (event.getAuthor().isBot()) return;
+        try {
+            if (!listenerService.IsUserExist(authorName)) {
+                listenerService.createDiscordUser(authorName);
+            }
```

Comment:
Use the `UserService` for this action. (See the comment in the `ListenerService`).